sync/class.jetpack-sync-queue.php

```diff
@@ -325,18 +325,62 @@ function unlock() {
 	}
 
 	private function get_checkout_id() {
-		return get_transient( $this->get_checkout_transient_name() );
+		global $wpdb;
+		$checkout_value = $wpdb->get_var( 
+			$wpdb->prepare(
+				"SELECT option_value FROM $wpdb->options WHERE option_name = %s", 
+				$this->get_lock_option_name()
+			)
+		);
+
+		if ( $checkout_value ) {
+			list( $checkout_id, $timestamp ) = explode( ':', $checkout_value );
+			if ( intval( $timestamp ) > time() ) {
+				return $checkout_id;
+			}
+		}
+
+		return false;
 	}
 
 	private function set_checkout_id( $checkout_id ) {
-		return set_transient( $this->get_checkout_transient_name(), $checkout_id, 5 * 60 ); // 5 minute timeout
+		global $wpdb;
+
+		$expires = time() + Jetpack_Sync_Defaults::$default_sync_queue_lock_timeout;
+		$updated_num = $wpdb->query(
+			$wpdb->prepare(
+				"UPDATE $wpdb->options SET option_value = %s WHERE option_name = %s", 
+				"$checkout_id:$expires",
+				$this->get_lock_option_name()
+			)
+		);
+
+		if ( ! $updated_num ) {
+			$updated_num = $wpdb->query(
+				$wpdb->prepare(
+					"INSERT INTO $wpdb->options ( option_name, option_value ) VALUES ( %s, %s )", 
+					$this->get_lock_option_name(),
+					"$checkout_id:$expires"
+				)
+			);
+		}
+
+		return $updated_num;
 	}
 
 	private function delete_checkout_id() {
-		delete_transient( $this->get_checkout_transient_name() );
+		global $wpdb;
+		// rather than delete, which causes fragmentation, we update in place
+		$wpdb->query( 
```

Comment:
Because it turns out updates in place are typically faster than inserts+deletes in MySQL, if you expect to be doing them a lot.